bin/docserv-createconfig

```diff
@@ -267,6 +267,21 @@ def check_langs(langs, message, attempts = 3):
     sys.exit("Expected one or more of these:\n"
         "{}".format(", ".join(languages)))
 
+def check_productid(productid, message, attempts = 3):
+    productid_regex = r'(?P<productid>[-_.+a-zA-Z0-9]+)'
```

Comment:
I'm wondering whether Stefan and I did this regex correct. Eli extracted it from our RNC schema.

@sknorr Maybe we've talked about this already, so sorry for the deja vu. :wink: 
I've read the comment in the RNC file ("An ID, however, we do not check for uniqueness here...") and I have the feeling the regex could be improved. I mean, the `productid` could start with a number, but a ID cannot. Shouldn't we consistent here (although we don't use uniqueness)?

According to the [W3C xml:id spec](https://www.w3.org/TR/xml-id/#processing), the attribute uses a NCName (a Name without the colon) which is defined in the XML 1.0 specification as:

```
[4] | NCName | ::= | Name - (Char* ':' Char*) | /* An XML Name, minus the ":" */
```

(`NAME` allows only letters as the first character, but no digits.)

So my question is:
Could we change the above regex (both in the RNC file and in the script) as follows:

```suggestion
    productid_regex = r'(?P<productid>[_a-zA-Z][-_.+a-zA-Z0-9]+)'
```

This would allow only letters as a first character. Example session:

```python
>>> productid_regex = r'(?P<productid>[_a-zA-Z][-_.+a-zA-Z0-9]+)'
>>> re.match(productid_regex, "123")
# => no match
>>> re.match(productid_regex, "abc123")
# => match
```

We don't want to start a product ID with a number, do we?

Reply:
We **absolutely need IDs that start with numbers** -- we use the same "ID" data type for our docset IDs and those IDs are usually a version number, e.g. "12-SP4" or "1.0". Since these setid values appear in the URL, I also won't accept the argument that "v12-SP4" would work just as well :). `doc.suse.com/sles/v12-SP4/...` is a clear regression compared to `doc.suse.com/sles/12-SP4/...`.

In essence, I consider your suggestion harmful.

In addition, when I say, "not checked for uniqueness here" what I meant was that we check for uniqueness via the docserv-stitch script but not plainly via the RNC. (Which is why I am so opposed to you trying to use the RNC to validate things as the RNC is only half the validation story.)